v2/ci.py

```diff
@@ -113,9 +113,19 @@ def _runTests(self):
                                                                                              })
         return subprocess.call(cmd, cwd=utils.get_k8s_folder())
 
+    def _collectWindowsLogs(self):
+        collect_logs_script="/tmp/k8s-ovn-ovs/v2/collect-logs.ps1"
+        for vm in self._get_windows_vms():
+            logs_vm_path = os.path.join(opts.log_path, "%s.zip" %(vm["name"]))
+            self.logging.info("Copying collect-logs script to node %s" % vm["name"])
+            self._copyTo(collect_logs_script, "c:\\", vm["name"], windows=True)
+            self._runRemoteCmd("c:\\collect-logs.ps1 -Destination C:\\k", vm["name"], windows=True)
+            self._copyFrom("C:\\k\\logs.zip", logs_vm_path, vm["name"], windows=True)
+
     def test(self):
         self._prepareTestEnv()
         self._prepareTests()
         ret = self._runTests()
+        self._collectWindowsLogs()
```

Comment:
Log collection should be a phase on it's own, as we would like to collect logs even if testing isn't enabled.  Say we just want --up, --down and we still want logs. Or say we want --up , --down, --test and --up fails: we would still want to try to get logs every possible log item.

Not to mention that even the way it's set now, if an exception happens along _prepareTestEnv or _prepareTests or _runTests() , log collection will never occur.